cmd/create/machinepool/cmd.go

```diff
@@ -400,6 +419,51 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 	}
 
+	isSpotSet := cmd.Flags().Changed("spot")
+	isSpotMaxPriceSet := cmd.Flags().Changed("spot-max-price")
+
+	spot := args.spotEnabled
+	spotMaxPrice := args.spotMaxPrice
+	if isSpotMaxPriceSet && isSpotSet && !spot {
+		reporter.Errorf("Can't set max price when spot is not enabled")
+		os.Exit(1)
+	}
+
+	if isSpotMaxPriceSet {
+		isSpotSet = true
+		spot = true
+	}
+
+	if !isSpotSet && interactive.Enabled() {
+		spot, err = interactive.GetBool(interactive.Input{
+			Question: "Use spot instances",
+			Help:     cmd.Flags().Lookup("spot").Usage,
+			Default:  spot,
+			Required: false,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid value for spot: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	if spot && !isSpotMaxPriceSet && interactive.Enabled() {
```

Comment:
I think this should be `if spot && interactive.Enabled() {` to give users the option of modifying the value during interactive mode.

Reply:
I disagree. It's a single command - why would the user want to set if in flag and then change it interactively?

Also - I don't want to override what the user set it they did, and I don't want to send a default if they didn't set anything

Reply:
Fair point, let's go with that for now. Currently there is a bit of inconsistency on that front: some commands show interactive fields _even_ if the user set the values, some commands hide them. I'll need to make that consistent at some point. Future improvements...

Reply:
I see that. Agreed